Source/SIMPLib/DataContainers/DataContainerProxy.h

```diff
@@ -35,141 +35,203 @@
+  DataContainerProxy& operator=(DataContainerProxy&&) = delete; // Move Assignment Not Implemented
+
+  /**
+   * @brief Returns the appropriate flag for the geometry type
+   * @param geoType The data container geometry type
+   * @return
+   */
+  static DataContainerProxy::DCGeometryTypeFlag GeometryTypeToFlag(IGeometry::Type geoType);
+
+  /**
+   * @brief operator = method
+   */
+  DataContainerProxy& operator=(const DataContainerProxy& amp);
+
+  /**
+   * @brief operator == method
+   */
+  bool operator==(const DataContainerProxy& amp) const;
+
+  /**
+   * @brief Writes the contents of the proxy to the json object 'json'
+   * @param json
+   * @return
+   */
+  void writeJson(QJsonObject& json);
+
+  /**
+   * @brief Reads the contents of the the json object 'json' into the proxy
+   * @param json
+   * @return
+   */
+  bool readJson(QJsonObject& json);
+
+  /**
+   * @brief Get the attribute matrices (Python Binding)
+   */
+  QMap<QString, AttributeMatrixProxy> getAttributeMatricies();
+
+  /**
+   * @brief Get the attribute matrices(Python Binding)
+   * @param new AttributeMatrices map
+   */
+  void setAttributeMatricies(const QMap<QString, AttributeMatrixProxy>&);
+
+  /**
+   * @brief Get the data container name (Python Binding)
+   */
+  QString getName() const;
+
+  /**
+   * @brief Set the data container name (Python Binding)
+   * @param new data container name
+   */
+  void setName(const QString& aName);
+
+  /**
+   * @brief Get the data container flag (Python Binding)
+   */
+  uint8_t getFlag() const;
+
+  /**
+   * @brief Set the data container flag (Python Binding)
+   * @param new data container flag
+   */
+  void setFlag(uint8_t flag);
+
+  /**
+   * @brief setDCType
+   * @param dType
+   */
+  void setDCType(uint32_t dType);
+
+  /**
+   * @brief getDCType
+   * @return
+   */
+  uint32_t getDCType() const;
+
+  /**
+   * @brief Toggle the data container flag (Python Binding)
+   */
+  void toggleFlag();
+
+  /**
+   * @brief Get the Attribute Matrix proxy
+   * @param the name of the attribute matrix
+   * @return the attribute matrix proxy
+   */
+  AttributeMatrixProxy& getAttributeMatrixProxy(const QString& name);
+
+  /**
+   * @brief setFlags
+   * @param flag
+   * @param amTypes
+   * @param primitiveTypes
+   * @param compDimsVector
+   */
+  void setFlags(uint8_t flag, AttributeMatrixProxy::AMTypeFlags amTypes = AttributeMatrixProxy::Any_AMType, DataArrayProxy::PrimitiveTypeFlags primitiveTypes = DataArrayProxy::Any_PType,
+                const DataArrayProxy::CompDimsVector& compDimsVector = DataArrayProxy::CompDimsVector());
+
+  /**
+   * @brief Updates the proxy to match a renamed DataArrayPath
+   * @param renamePath
+   */
+  void updatePath(DataArrayPath::RenameType renamePath);
+
+  //----- Our variables, publicly available
+  uint8_t flag;
+  QString name;
+  unsigned int dcType;
+  QMap<QString, AttributeMatrixProxy> attributeMatricies;
```

Comment:
We have getters and setters for all of these variables, and SIMPL code has been updated to use them.  Do we still want these to be public variables, or private?

Reply:
Some of the codes still directly access them and I have not had the time to work through the ramifications of making them private (which I agree that we should do).